### PR TITLE
Pass the receiver to get_property_on_prototype (fixes #11600)

### LINKED ISSUE #11600: Throw when accessing IDL members on arbitrary objects with platform prototypes
See the test tests/wpt/web-platform-tests/dom/collections/HTMLCollection-as-proto-length-get-throws.html.

We probably don't implement the various "if O is not a platform object that implements interface I, throw a TypeError" steps that [the WebIDL spec](http://heycam.github.io/webidl/#es-interfaces) defines.

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13017)

### CODE REVIEW

**Comment by Ms2ger** [comment]
r+ if it passes

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -4529,10 +4529,17 @@ def getBody(self):
...
-                        "if RUST_JSID_IS_STRING(id) && !has_property_on_prototype(cx, proxy, id) {\n" +
-                        CGIndenter(CGProxyNamedGetter(self.descriptor, templateValues)).define() + "\n" +
-                        "}\n")
+            namedGet = """
+if RUST_JSID_IS_STRING(id) {
+    let mut has_on_proto = false;
+    if !has_property_on_prototype(cx, proxy, id, &mut has_on_proto) {
+        return false;
+    }
+    if has_on_proto {
```
Comment (jdm) [inline components/script/dom/bindings/codegen/CodegenRust.py]: This should be `if !has_on_proto`

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 2]
```diff
@@ -4757,12 +4764,20 @@ def getBody(self):
...
-                     "    return true;\n"
-                     "}\n" +
-                     "\n")
+            named = """\
+if RUST_JSID_IS_STRING(id) {
+    let mut has_on_proto = false;
+    if !has_property_on_prototype(cx, proxy, id, &mut has_on_proto) {
+        return false;
+    }
+    if has_on_proto {
```
Comment (jdm) [inline components/script/dom/bindings/codegen/CodegenRust.py]: This should be `if !has_on_proto`

Comment (nox) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Ah right, thanks.

**Comment by jdm** [comment]
I am surprised that we don't have new tests passing if this actually fixes #11600.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 3]
```diff
@@ -4952,12 +4959,20 @@ def getBody(self):
...
-                     CGIndenter(CGProxyNamedGetter(self.descriptor)).define() + "\n" +
-                     "    *bp = found;\n"
-                     "    return true;\n"
-                     "}\n" +
-                     "\n")
+            named = """\
+if RUST_JSID_IS_STRING(id) {
+    let mut has_on_proto = false;
+    if !has_property_on_prototype(cx, proxy, id, &mut has_on_proto) {
+        return false;
```
Comment (emilio) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Shouldn't this set `*bp = false`? (sorry if it's a dumb question, I was digging into the reason for #13165 and arrived here).

**Pull request merged**